File: backend/src/main/java/com/gte619n/healthfitness/core/trt/TrtContext.java
package com.gte619n.healthfitness.core.trt;

import java.util.List;

/**
 * The grounded TRT context for a designer turn: whether the user is on TRT,
 * the current monitoring-panel markers (latest value + trend + status), and
 * any danger flags fired this turn (ADR-0015).
 *
 * <p>Component names are part of the client wire contract — do not rename.
 */
public record TrtContext(
    boolean onTrt,
    List<TrtMarker> markers,
    List<DangerFlag> dangerFlags
) {}

File: android/core-domain/src/main/java/com/gte619n/healthfitness/domain/workouts/trt/TrtContext.kt
package com.gte619n.healthfitness.domain.workouts.trt

import java.time.LocalDate

// Grounded TRT decision-support context surfaced in the workout-program designer
// chat (IMPL-AND-18 / ADR-0015). Mirrors the LOCKED backend
// `GET api/me/workout-programs/chat/trt-context` shape: monitoring-panel markers
// vs. range with trend + status, plus any hard danger flags.

/** Direction of a marker across the two most-recent reports. */
enum class TrtTrend { RISING, FALLING, STABLE, UNKNOWN }

/** Where a marker's latest value sits relative to its reference range. */
enum class TrtMarkerStatus { LOW, IN_RANGE, HIGH, WATCH, UNKNOWN }

/** Severity of a danger flag (S6e mandatory hard alerts). */
enum class DangerSeverity { WARNING, DANGER }

data class TrtMarker(
    val name: String,
    val label: String,
    val value: Double?,
    val unit: String?,
    val refLow: Double?,
    val refHigh: Double?,
    val sampleDate: LocalDate?,
    val trend: TrtTrend,
    val status: TrtMarkerStatus,
)

data class DangerFlag(
    val marker: String,
    val severity: DangerSeverity,
    val message: String,
)

data class TrtContext(
    val onTrt: Boolean,
    val markers: List<TrtMarker> = emptyList(),
    val dangerFlags: List<DangerFlag> = emptyList(),
) {
    /** Show the TRT panel only when on TRT or there is at least one marker. */
    val shouldShow: Boolean
        get() = onTrt || markers.isNotEmpty()
}
